src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,373 @@
+            {
+                // This will make sure there is no duplicate 'await'
+                // Example:
+                // Before:
+                // async Task Caller() => await Callee();
+                // async Task Callee() => await Task.CompletedTask;
+                // After:
+                // async Task Caller() => await Task.CompletedTask;
+                // async Task Callee() => await Task.CompletedTask;
+                // The original inline expression in callee will be 'await Task.CompletedTask'
+                // The caller just need 'Task.CompletedTask' without the 'await'
+                inlineExpression = _syntaxFacts.GetExpressionOfAwaitExpression(inlineExpression) as TExpressionSyntax;
+            }
+
+            if (inlineExpression == null)
+            {
+                return;
+            }
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var callerDeclarationNode = _syntaxFacts.GetContainingMemberDeclaration(root, calleeMethodInvocationNode.SpanStart)
+                ?? GetEnclosingMethodLikeNode(calleeMethodNode);
+            if (callerDeclarationNode == null)
+            {
+                return;
+            }
+
+            var callerSymbol = semanticModel.GetDeclaredSymbol(callerDeclarationNode, cancellationToken)
+                ?? semanticModel.GetSymbolInfo(callerDeclarationNode, cancellationToken).GetAnySymbol();
+            if (callerSymbol == null)
+            {
+                return;
+            }
+
+            var containsAwaitExpression = ContainsAwaitExpression(inlineExpression, calleeMethodNode);
+            if (containsAwaitExpression)
+            {
+                // If there is nested 'await', check if the caller is a method, and its return type
+                // is void or AwaitableType because after move the 'await' to the caller it must be async
+                // Example that shouldn't offer refactoring:
+                // int Caller()
+                // {
+                //    var x = Callee();
+                //    return 1;
+                // }
+                // async Task Callee() => await Task.Delay(await Task.FromResult(100));
+                // Example that should offer refactoring (because it is safe to change the caller to async):
+                // void Caller()
+                // {
+                //    var x = Callee();
+                // }
+                // async Task Callee() => await Task.Delay(await Task.FromResult(100));
+                var canInlineAwaitExpression = callerSymbol is IMethodSymbol callerMethodSymbol
+                       && (callerMethodSymbol.IsAsync ||
+                           callerMethodSymbol.ReturnsVoid ||
+                           calleeMethodSymbol.IsAwaitableNonDynamic(semanticModel, callerDeclarationNode.SpanStart));
+                if (!canInlineAwaitExpression)
```

Comment:
i have to think on this.

Reply:
I have thought another 2 ways for this,
1. Just put the await expression to the caller, which will cause error, the user needs to fix it.
2. We would try to convert the return type.
Like
`int Caller()
{
      var x = Callee();
      return 1;
}`
`async Task Callee() => await Task.Delay(await Task.FromResult(100));`
The return type could be changed to 'Task<int>' if the user wants to do the refactoring. (I am guessing Task<T> can be fetched from Compiliation)
But it might  not be accurate still, because user might want ValueTask.

Reply:
As discussed offline, the scenario here would be
if the caller is a method that has a awaitable return type, then add 'async'
For All the other cases, just put the 'await' expression in the method body. (let user to fix it)